gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java

```diff
@@ -83,21 +83,20 @@
   private long duration;
 
   // Needed for serialization/deserialization
-  public TaskState() {
-  }
+  public TaskState() {}
 
   public TaskState(WorkUnitState workUnitState) {
     // Since getWorkunit() returns an immutable WorkUnit object,
     // the WorkUnit object in this object is also immutable.
-    super(workUnitState.getWorkunit());
+    super(workUnitState.getWorkunit(), workUnitState.getJobState());
     addAll(workUnitState);
```

Comment:
We should also fix that so that a `TaskState` only has a pointer to its `WorkUnitState`, rather than making a full copy of all the properties in the `WorkUnitState`.

On an unrelated note, whats the difference between `TaskState` and `WorkUnitState`? I don't see any functional reason why the two can't be merged, they both seem to try to track some runtime information. But thats a discussion for another day.